physical/postgresql/postgresql.go

```diff
@@ -236,6 +237,57 @@ func NewPostgreSQLBackend(conf map[string]string, logger log.Logger) (physical.B
 		upsert_function: quoted_upsert_function,
 	}
 
+	// Determine if we should create tables.
+	raw_skip_create_table, ok := conf["skip_create_table"]
+	if !ok {
+		raw_skip_create_table = "false"
+	}
+	skip_create_table, err := parseutil.ParseBool(raw_skip_create_table)
+	if err != nil {
+		return nil, fmt.Errorf("failed to parse value for `skip_create_table`: %w", err)
+	}
+	if !skip_create_table {
+		// Try a query and see if there's a table not found error for both
+		// tables: data and ha locks.
+		var path string
+		row := db.QueryRow("SELECT path FROM " + quoted_table + " LIMIT 1")
+		if err := row.Scan(&path); err != nil && strings.Contains(err.Error(), `relation `+quoted_table+` does not exist`) {
+			// Successfully detected that there is no table; create it.
+			createTableQuery := "CREATE TABLE " + quoted_table + " (" +
```

Comment:
Thoughts on if this should just be a `CREATE TABLE IF NOT EXISTS` instead?

Reply:
Technically I think if we wrap these in a single transaction, we could unconditionally issue them and better handle multiple nodes starting at the same time with the same database.